### PR TITLE
Temporarily adds CURRENT_USER to Sharify Global Data 

### PR DESCRIPTION
**Pull request by ashleyjelks**
This PR temporarily adds the CURRENT_USER field to the Sharify Global Data object. Currently Typescript throws an error because CURRENT_USER isn't a property of the Global Data object. To work around this we have to use Lodash's Get method, which disallows the other type checking benefits of Typescript. As we only need the CURRENT_USER to check if a CURRENT_USER is either an Admin or Allowlisted user to hide the new externally hosted ads, this change is temporary. Once the new ad implementation is production ready, we won't need to hide it behind these checks or feature flags.

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 72e32c7fae18b4078bf8f98dfe36d2b234131cce

https://deploy-preview-2397--artsy-reaction.netlify.com

**Comment by damassi** [comment]
@ashleyjelks Can you add content to the description describing this change?

File: typings/sharify.d.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ declare module "sharify" {
       readonly APP_URL: string
       readonly ARTIST_COLLECTIONS_RAIL?: string // TODO: remove after CollectionsRail a/b test
       readonly ARTIST_COLLECTIONS_RAIL_IDS: string[]
+      readonly CURRENT_USER?: User
```
Comment (damassi) [inline typings/sharify.d.ts]: Will want to add a `// FIXME: Remove after ...` here

File: src/Components/Publishing/Ads/EnabledAd.ts [hunk 2]
```diff
@@ -12,10 +11,9 @@ export const isHTLAdEnabled = () => {
   const allowedUsers = (sd.HASHTAG_LAB_ADS_ALLOWLIST || "")
     .split(",")
     .filter(Boolean)
-  const currentUser = get(sd, "CURRENT_USER.email", "")
+  const currentUser = sd.CURRENT_USER.email
   const isAllowedUser = allowedUsers.includes(currentUser)
-  const isAdminUser =
-    get(sd, "CURRENT_USER.type", "") === "Admin" ? true : false
+  const isAdminUser = sd.CURRENT_USER.type === "Admin" ? true : false
```
Comment (damassi) [inline src/Components/Publishing/Ads/EnabledAd.ts]:
```
const isAdminUser = sd.CURRENT_USER.type === "Admin" || false
```

File: .storybook/webpack.config.js [hunk 3]
```diff
@@ -19,6 +19,7 @@ const {
   CI,
   CMS_URL,
   APP_URL,
+  CURRENT_USER,
```
Comment (damassi) [inline .storybook/webpack.config.js]: `// FIXME: Remove when ...`

File: .storybook/webpack.config.js [hunk 4]
```diff
@@ -50,6 +51,7 @@ const notOnCI = value => (isCI ? [] : [value])
 const sharifyPath = sharify({
   APP_URL,
   CMS_URL,
+  CURRENT_USER,
```
Comment (damassi) [inline .storybook/webpack.config.js]: Same here

**Comment by artsyit** [comment]
:rocket: PR was released in v16.10.2 :rocket:

**Pull request merged**